.github/workflows/main.yml

```diff
@@ -24,12 +45,29 @@ jobs:
     runs-on: ${{ matrix.platform }}
     steps:
       - uses: actions/checkout@v2
-      - name: Setup Python
-        uses: actions/setup-python@v1
+      - uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python }}
-      - name: "Install dependencies"
+      - name: Install dependencies
         run: |
           python -m pip install --upgrade pip setuptools wheel tox
       - name: Run tests
-        run: python -m tox -e py  # Run tox using the version of Python in `PATH`
+        run: python -m tox -e py -- --cov-report xml
+      - uses: codecov/codecov-action@v1
+        with:
+          file: ./coverage.xml
+          name: ${{ matrix.python }} - ${{ matrix.platform }}
+          fail_ci_if_error: true
+
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip setuptools wheel tox
```

Comment:
Can we add `.github/actions/requirements.txt` with these and run `python -m pip install --upgrade -r .github/actions/requirements.txt` instead? That way all of these are less duplicated? Helps when we move to something else and might be a decent pattern to upstream to the python action

Reply:
I like that idea, but it got me thinking that it would be sufficient (if not preferred) to let `tox` manage the versions of `pip`, `setuptools`, and `wheel`, so in https://github.com/pypa/twine/pull/705/commits/7c65ce6d853d96784c0020cf56de6e8362ca7398 this has been reduced to `pip install tox`.

The [output of `pip list`](https://github.com/pypa/twine/runs/1237910465#step:5:13) shows that the testenvs are using the latest versions,  but if we want more control in the future, we can use the [`requires`](https://tox.readthedocs.io/en/latest/config.html#conf-requires) and [`download`](https://tox.readthedocs.io/en/latest/config.html#conf-download) options in `tox.ini` (with the caveat that there's an [open issue re: `download = true`](https://github.com/tox-dev/tox/issues/1558).